### PR TITLE
set max page size for changefeed

### PR DESCRIPTION
**Pull request by endophage**
Signed-off-by: David Lawrence <david.lawrence@docker.com> (github: endophage)

### CODE REVIEW

File: const.go [hunk 1]
```diff
@@ -74,6 +74,9 @@ const (
 
 	// DefaultPageSize is the default number of records to return from the changefeed
 	DefaultPageSize = 100
+	// MaxPageSize is the maximum number of records that can be requested from
+	//  the changefeed in a single request
```
Comment (HuKeping) [inline const.go]: nit: how about just one space between the `//` and `the`

File: server/handlers/changefeed.go [hunk 2]
```diff
@@ -77,9 +77,15 @@ func checkChangefeedInputs(logger ctxu.Logger, s interface{}, r string) (
...
 		}
-		if pageSize == 0 {
-			pageSize = notary.DefaultPageSize
-		}
+
+	}
+	switch {
+	case pageSize == 0:
+		pageSize = notary.DefaultPageSize
+	case pageSize > notary.MaxPageSize:
```
Comment (riyazdf) [inline server/handlers/changefeed.go]: can we add tests to check for `MaxPageSize` and `-MaxPageSize`?

Comment (endophage) [inline server/handlers/changefeed.go]: Definitely, was planning to. @cyli asked for them while I was on my way out of the office last night :)

**Comment by endophage** [comment]
@HuKeping I'm also going to do you suggestion about having `checkChangefeedInputs` return an `int` rather than an `int64`

**Comment by HuKeping** [comment]
Thanks for making that change @endophage !

**Comment by endophage** [comment]
@cyli @riyazdf can we get another LGTM?

**Pull request merged**